### PR TITLE
chore(instrumentation-pg): Inline incubating constants from @opentelemetry/semantic-conventions

### PR DESCRIPTION
**Pull request by nwalters512**
## Which problem is this PR solving?

~~No other instrumentation packages pin `@opentelemetry/semantic-conventions`, and from what I can tell, there's no good reason that `@opentelemetry/instumentation-pg` should be pinning it. When the dependency is pinned, npm can't dedupe it to a higher version, which results in multiple versions of this package being installed.~~

As decided in https://github.com/open-telemetry/opentelemetry-js/issues/5182#issuecomment-2536558790, `incubating` constants from `@opentelemetry/semantic-conventions` should be inlined instead of pinning `@opentelemetry/semantic-conventions` to a particular version.

## Short description of the changes

~~In `@opentelemetry/instrumentation-pg`, `@opentelemetry/semantic-conventions` now uses a `^` semver range to allow newer minor versions to be installed.~~

This PR updates `@opentelemetry/instrumentation-pg` to include inlined copies of the experimental constants it needs. This, in turn, allows us to unpin `@opentelemetry/semantic-conventions`.

### CODE REVIEW

**Comment by maryliag** [comment]
The reason for this versioning being pinned, it's because this package is using the `incubating` entrypoint, which may break on minor bumps.
This was added on purpose here: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2473
Once it's no longer using the `incubating`, it can be unpinned.

I see other packages that use the `incubating` entrypoint are not pinning the value, like this [one](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-aws-lambda/package.json#L60).  @pichlermarc do you know if this was just a mistake or we can now unpin for these cases?

**Comment by nwalters512** [comment]
Ah! Apologies, I didn't go digging deep enough. In that case, can I just change this PR to pin to 1.28.0 so that it will be deduped by package managers (at least until 1.29.0 comes out)?

**Comment by pichlermarc** [comment]
> I see other packages that use the incubating entrypoint are not pinning the value, like this [one](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-aws-lambda/package.json#L60). @pichlermarc do you know if this was just a mistake or we can now unpin for these cases?

Hi 🙂 I'd say it's a mistake, when using the incubating entrypoint, we should always pin for now until we have a decision on https://github.com/open-telemetry/opentelemetry-js/issues/5182. Sorry for the inconvenience.

**Comment by maryliag** [comment]
Hi @nwalters512 , looks like we will go with a different approach of copying those specific values from the incubation, see more here: https://github.com/open-telemetry/opentelemetry-js/issues/5182#issuecomment-2536558790

I would be happy to review if that is something you want to work on.

**Comment by nwalters512** [comment]
@maryliag I'd be happy to open some PRs for this! I'll work on `instrumentation-pg` to start, and I can follow up with other packages that are using the `incubating` entrypoint.

**Comment by maryliag** [comment]
thank you @nwalters512!

File: plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ import {
   METRIC_DB_CLIENT_OPERATION_DURATION,
   ATTR_DB_NAMESPACE,
   ATTR_DB_OPERATION_NAME,
-} from '@opentelemetry/semantic-conventions/incubating';
+} from './semantic-conventions';
```
Comment (nwalters512) [inline plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts:68]: https://github.com/open-telemetry/opentelemetry-js/issues/5182#issuecomment-2536558790 suggested using `semconv.ts`; I went with `semantic-conventions.ts` since it directly matches the package name. Happy to use the other suggestion if that's strongly preferred.

File: plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/**
+ * These constants are considered experimental exports of `@opentelemetry/semantic-conventions`.
+ * They're being inlined until they're officially exported by `@opentelemetry/semantic-conventions`.
+ */
```
Comment (nwalters512) [inline plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts:4]: This little comment will appear at the top of this file in other packages; let's workshop it if needed before it proliferates.

File: plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * The number of connections that are currently in state described by the `state` attribute
+ *
+ * @experimental This metric is experimental and is subject to breaking changes in minor releases of `@opentelemetry/semantic-conventions`.
```
Comment (nwalters512) [inline plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts:9]: I copied these all in full, including the "experimental" warning. Happy to edit this to remove it desired.

**Review by nwalters512 — COMMENTED** [review]
@maryliag I repurposed this PR to include the inlining of the relevant constants.

**Comment by nwalters512** [comment]
Sorry about the failing tests, I'll get those fixed up.

**Comment by maryliag** [comment]
Thank you for working on this!

@trentm would you mind confirming if that is what you had in mind?

File: plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts (hunk 2 above)
Comment (trentm) [reply plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts:20]: FWIW, in a third-party instrumentation I have at work that is doing this, I have this block comment:
https://github.com/elastic/elastic-otel-node/blob/690896cb2995a349313b37ae6e609a741676946c/packages/instrumentation-openai/src/semconv.ts#L20-L27

That's perhaps a little wordier than necessary, though.

File: plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts (hunk 3 above)
Comment (trentm) [reply plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts:25]: I think either way is fine. 

(Ideally eventually there is some handy tooling for this to help with generation and maintenance of these. I'd played with a *start* at something, but it is far from ready for others to use:
https://github.com/elastic/elastic-otel-node/blob/main/packages/instrumentation-openai/scripts/semconv-gen.js)

File: plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts
Comment (trentm) [inline plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts:1]: nit: I have a small preference for "semconv.ts" if others think that is reasonable as the filename. (It is the name I mentioned as an example at https://github.com/open-telemetry/opentelemetry-js/blob/main/semantic-conventions/README.md#unstable-semconv)

I'm happy to be overridden by others' opinions on this. The current naming of `.ts` files in this repo *does* tend to use this-style.ts rather than using abbreviations in filenames.

**Review by trentm — APPROVED** [review]
> @trentm would you mind confirming if that is what you had in mind?

Yes, this is what I had in mind. Thanks!

I added a couple comments and a nit. Good to go as is, however.

File: plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts
Comment (nwalters512) [reply plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts:1]: I commented on this here too: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2599#discussion_r1880770481

I'll defer to whatever you or the other maintainers think is best 🙂

File: plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts (hunk 2 above)
Comment (nwalters512) [reply plugins/node/opentelemetry-instrumentation-pg/src/semantic-conventions.ts:20]: Agreed that it's perhaps a bit wordy. I'd be happy to apply any recommendations to the message I have here!

**Comment by trentm** [comment]
@nwalters512 https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2664 is doing the same thing. There was discussion on Slack to refine that one. I'm merging that one and closing this one. Thanks for the changes!
